lib/rspec/core/runner.rb

```diff
@@ -95,7 +102,7 @@ def run(err, out)
       # @param out [IO] output stream
       def setup(err, out)
         configure(err, out)
-        @configuration.load_spec_files
```

Comment:
Stylistically, we'd prefer a `return if RSpec.world.wants_to_quit` guard rather than this, we don't mind if the filters aren't announced.

Reply:
I copied the output from the `spec_helper` errors in other spec, and this is what it took to make the output the same.

Reply:
Its ok to change the output.

Reply:
I disagree with giving the user different outputs for the same error, just to satisfy a stylistic concern.

Reply:
Why output filter information if they're never going to apply? It just confuses matters.

Reply:
Oh would this change existing documented output? Hm that changes things, can we do the announce in an unsure?

Reply:
Yes. A single sentence for a single spec (which is the one I used as a base for the one I'm adding here). Maybe it's easier if I show the diff:

```diff
--- a/lib/rspec/core/runner.rb
+++ b/lib/rspec/core/runner.rb
@@ -102,7 +102,9 @@ module RSpec
       # @param out [IO] output stream
       def setup(err, out)
         configure(err, out)
-        @configuration.load_spec_files unless RSpec.world.wants_to_quit
+        return if RSpec.world.wants_to_quit
+
+        @configuration.load_spec_files
         @world.announce_filters
       end
 
diff --git a/spec/integration/spec_file_load_errors_spec.rb b/spec/integration/spec_file_load_errors_spec.rb
index f4a5d062..c2df37be 100644
--- a/spec/integration/spec_file_load_errors_spec.rb
+++ b/spec/integration/spec_file_load_errors_spec.rb
@@ -45,7 +45,6 @@ RSpec.describe 'Spec file load errors' do
       RuntimeError:
         boom
       # ./helper_with_error.rb:1#{spec_line_suffix}
-      No examples found.
 
 
       Finished in n.nnnn seconds (files took n.nnnn seconds to load)
@@ -76,7 +75,6 @@ RSpec.describe 'Spec file load errors' do
       RuntimeError:
         boom
       # ./helper_with_error.rb:1#{spec_line_suffix}
-      No examples found.
 
 
       Finished in n.nnnn seconds (files took n.nnnn seconds to load)
```

Reply:
In my opinion, you're right and announcing filters in this situation doesn't make much sense. What value does the "No examples found" add in this context?

Compatibility is another issue... Not sure how much trouble in terms of backwards compatibility this change could cause. I'd say not a lot, this output is not very parsable, so I'm not sure someone is relying on parsing it the way it is now?

Reply:
What about is we do:
```
  return if RSpec.world.wants_to_quit
  @configuration.load_spec_files
ensure
   @world.announce_filters
```

Reply:
So it seems that we have switched positions, now you think that we should not change any output (like I did initially), and I think that there's no need to announce filters in this situation (like you did initially)... :rofl:

Anyways, just to confirm, what benefit does your alternative provide over the current version? If the motivation is that it is more readable in your opinion, then I'll just make the changes, I just want to make sure I'm not missing anything.

Reply:
Its been our convention to prefer guard conditions like `return ... if ...` when exiting early, and also to avoid `unless`

Reply:
I find the "early return + ensure" style pretty obscure, and from a quick scan I couldn't find other examples in rspec-core's code (there might be some, though). Regarding `unless`, it seems to be quite common in the current code base to prefer "unless + positive condition" to "if + negated positive condition".

Anyways, I just wanted to confirm this was just a matter of style, I'll change it to the way you like it!

Reply:
As you might have gathered, we don't change existing code to enforce styles, partly through "less churn" partly through lack of time!